src/main/scala/com/databricks/labs/overwatch/env/Database.scala

```diff
@@ -248,82 +341,25 @@ class Database(config: Config) extends SparkSessionWrapper {
-      } else ""
-      val mergeCondition: String = immutableColumns.map(k => s"updates.$k = target.$k").mkString(" AND ") + " " +
-        s"AND target.organization_id = '${config.organizationId}'" + // force partition filter for concurrent merge
-        explicitDatePartitionCondition // force right side scan to only scan relevant dates
-
-      val mergeDetailMsg =
-        s"""
-           |Beginning upsert to ${target.tableFullName}.
-           |MERGE CONDITION: $mergeCondition
-           |""".stripMargin
-      logger.log(Level.INFO, mergeDetailMsg)
-      spark.conf.set("spark.databricks.delta.commitInfo.userMetadata", config.runID)
-      // TODO -- when DBR 9.1 LTS GA, use LSM (low-shuffle-merge) to improve pipeline
-      deriveDeltaMergeBuilder(deltaTarget, updatesDF, mergeCondition, target)
-        .execute()
-
-      spark.conf.unset("spark.databricks.delta.commitInfo.userMetadata")
-
-    } else {
-      logger.log(Level.INFO, s"Beginning write to ${target.tableFullName}")
-      if (target.checkpointPath.nonEmpty) { // STREAMING WRITER
-
-        val msg = s"Checkpoint Path Set: ${target.checkpointPath.get} - proceeding with streaming write"
-        logger.log(Level.INFO, msg)
-        if (config.debugFlag) println(msg)
-
-        val beginMsg = s"Stream to ${target.tableFullName} beginning."
-        if (config.debugFlag) println(beginMsg)
-        logger.log(Level.INFO, beginMsg)
-        if (!spark.catalog.tableExists(config.databaseName, target.name)) {
-          initializeStreamTarget(finalDF, target)
-        }
-        val streamWriter = target.writer(finalDF)
-          .asInstanceOf[DataStreamWriter[Row]]
-          .option("path", target.tableLocation)
-          .start()
-        val streamManager = getQueryListener(streamWriter, config.auditLogConfig.azureAuditLogEventhubConfig.get.minEventsPerTrigger)
-        spark.streams.addListener(streamManager)
-        val listenerAddedMsg = s"Event Listener Added.\nStream: ${streamWriter.name}\nID: ${streamWriter.id}"
-        if (config.debugFlag) println(listenerAddedMsg)
-        logger.log(Level.INFO, listenerAddedMsg)
-
-        streamWriter.awaitTermination()
-        spark.streams.removeListener(streamManager)
-
-      } else { // DF Standard Writer append/overwrite
-        try {
-          target.writer(finalDF).asInstanceOf[DataFrameWriter[Row]].save(target.tableLocation)
-        } catch {
-          case e: Throwable =>
-            val fullMsg = PipelineFunctions.appendStackStrace(e, "Exception in writeMethod:")
-            logger.log(Level.ERROR, s"""Exception while writing to ${target.tableFullName}"""" + Thread.currentThread().getName + " Error msg:" + fullMsg)
-            throw e
-        }
-      }
-      logger.log(Level.INFO, s"Completed write to ${target.tableFullName}")
+    target.writeMode match {
+      case WriteMode.merge => batchWriter(finalDF, target, maxMergeScanDates)
+      case _ => streamWriter(finalDF, target)
```

Comment:
uhhh soo what about WriteMode.append and overwrite? Below's a screenshot of what used to handle it.

You have the following conditions: 
* merge
* Not Merge -- this can either be
  * Stream
  * Append/Overwrite

![image](https://github.com/databrickslabs/overwatch/assets/10840635/e4307436-0d2e-4cad-8ac5-8cbbb68d2029)

Reply:
@GeekSheikh This else block is covered in streamWriter function.